lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -897,58 +893,94 @@ def get_vm_network_interfaces(self, vm=None):
 
         return device_list
 
-    def configure_network(self, vm_obj):
-        # Ignore empty networks, this permits to keep networks when deploying a template/cloning a VM
-        if len(self.params['networks']) == 0:
-            return
+    def sanitize_network_params(self):
+        """
+        Function to sanitize user provided network provided params
+
+        Returns: A sanitized list of network params, else fails
 
+        """
         network_devices = list()
+        # Clean up user data here
         for network in self.params['networks']:
-            if 'ip' in network or 'netmask' in network:
-                if 'ip' not in network or 'netmask' not in network:
-                    self.module.fail_json(msg="Both 'ip' and 'netmask' are required together.")
-
-            if 'name' in network:
-                if find_obj(self.content, [vim.Network], network['name']) is None:
-                    self.module.fail_json(msg="Network '%(name)s' does not exists" % network)
+            if 'name' not in network and 'vlan' not in network:
+                self.module.fail_json(msg="Please specify at least a network name or"
+                                          " a VLAN name under VM network list.")
 
+            if 'name' in network and find_obj(self.content, [vim.Network], network['name']) is None:
+                self.module.fail_json(msg="Network '%(name)s' does not exists" % network)
             elif 'vlan' in network:
                 dvps = self.cache.get_all_objs(self.content, [vim.dvs.DistributedVirtualPortgroup])
                 for dvp in dvps:
-                    if hasattr(dvp.config.defaultPortConfig, 'vlan') and dvp.config.defaultPortConfig.vlan.vlanId == network['vlan']:
+                    if hasattr(dvp.config.defaultPortConfig, 'vlan') and \
+                            dvp.config.defaultPortConfig.vlan.vlanId == network['vlan']:
                         network['name'] = dvp.config.name
                         break
                     if dvp.config.name == network['vlan']:
                         network['name'] = dvp.config.name
                         break
                 else:
-                    self.module.fail_json(msg="VLAN '%(vlan)s' does not exist" % network)
+                    self.module.fail_json(msg="VLAN '%(vlan)s' does not exist." % network)
+
+            if 'type' in network:
+                if network['type'] not in ['dhcp', 'static']:
+                    self.module.fail_json(msg="Network type '%(type)s' is not a valid parameter."
+                                              " Valid parameters are ['dhcp', 'static']." % network)
+                if network['type'] != 'static':
+                    if 'ip' in network or 'netmask' in network:
```

Comment:
`if network['type'] != 'static' and ( 'ip' in network or 'netmask' in network):`

Reply:
Done